src/iidm/OperationalLimitsHolder.cpp

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <powsybl/iidm/OperationalLimitsHolder.hpp>
+
+#include <boost/range/adaptor/indirected.hpp>
+#include <boost/range/adaptor/map.hpp>
+
+#include <powsybl/iidm/ActivePowerLimitsAdder.hpp>
+#include <powsybl/iidm/ApparentPowerLimitsAdder.hpp>
+#include <powsybl/iidm/CurrentLimitsAdder.hpp>
+#include <powsybl/iidm/Identifiable.hpp>
+
+namespace powsybl {
+
+namespace iidm {
+
+OperationalLimitsHolder::OperationalLimitsHolder(Identifiable& identifiable, std::string attributeName) :
+    m_identifiable(identifiable),
+    m_attributeName(std::move(attributeName)) {
+}
+
+std::string OperationalLimitsHolder::getMessageHeader() const {
+    return m_identifiable.getMessageHeader();
+}
+
+stdcxx::const_range<OperationalLimits> OperationalLimitsHolder::getOperationalLimits() const {
+    return m_operationalLimits | boost::adaptors::map_values | boost::adaptors::indirected;
+}
+
+stdcxx::range<OperationalLimits> OperationalLimitsHolder::getOperationalLimits() {
+    return m_operationalLimits | boost::adaptors::map_values | boost::adaptors::indirected;
+}
+
+ActivePowerLimitsAdder OperationalLimitsHolder::newActivePowerLimits() {
+    return ActivePowerLimitsAdder(*this);
+}
+
+ApparentPowerLimitsAdder OperationalLimitsHolder::newApparentPowerLimits() {
+    return ApparentPowerLimitsAdder(*this);
+}
+
+CurrentLimitsAdder OperationalLimitsHolder::newCurrentLimits() {
+    return CurrentLimitsAdder(*this);
+}
+
+void OperationalLimitsHolder::setOperationalLimits(const LimitType& limitType, std::unique_ptr<OperationalLimits>&& operationalLimits) {
+    m_operationalLimits[limitType] = std::move(operationalLimits);
+}
```

Comment:
I fixed the implementation in the Java code according to my discussion with @MioRtia. If you have no limits, and you try to remove each kind of limits, at the end you have a map with null values in it.

The java implementation is:
```
    @Override
    public void setOperationalLimits(LimitType limitType, OperationalLimits operationalLimits) {
        OperationalLimits oldValue;
        if (operationalLimits == null) {
            oldValue = this.operationalLimits.remove(limitType);
        } else {
            oldValue = this.operationalLimits.put(limitType, operationalLimits);
        }
        identifiable.getNetwork().getListeners().notifyUpdate(identifiable, attributeName + "_" + limitType, oldValue, operationalLimits);
    }
```

Note that this will have an impact on `getOperationLimits` implementation cause the operationalLimit could not be null anymore.

Reply:
>Note that this will have an impact on getOperationLimits implementation cause the operationalLimit could not be null anymore.

Do you mean that we throw an exception is users tries to get a limit from template type which doesn't exist ? Unless I'm wrong, such a case returns null in Java.